File: Microservice1/src/main/java/com/example/demo/services/AdminService.java
package com.example.demo.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.demo.entities.User;
import com.example.demo.repositories.UserRepo;

@Service
public class AdminService {

    @Autowired
    private UserRepo repo;

    // Pending landlords
    public List<User> getPendingLandlords() {
        return repo.findByRoleRidAndStatus(3, false);
    }

    // Approve landlord
    public User approveLandlord(int uid) {

        User user = repo.findById(uid).orElse(null);

        if (user != null) {
            user.setStatus(true);
            return repo.save(user);
        }

        return null;
    }

    // Reject landlord
    public void rejectLandlord(int uid) {
        repo.deleteById(uid);
    }

}

File: Frontend/src/services/adminService.js
import { adminApi } from "./api";

// Matches com.example.demo.controller.AdminController on
// Microservice 1 (Rental-management), reached through the gateway.

const adminService = {
  // GET /admin/landlords/pending
  async getPendingLandlords() {
    const res = await adminApi.get("/admin/landlords/pending");
    return res.data;
  },

  // PUT /admin/approve/{uid}
  async approveLandlord(uid) {
    const res = await adminApi.put(`/admin/approve/${uid}`);
    return res.data;
  },

  // DELETE /admin/reject/{uid}
  async rejectLandlord(uid) {
    const res = await adminApi.delete(`/admin/reject/${uid}`);
    return res.data;
  },
};

export default adminService;
